### PR TITLE
chore: add missing error codes for gmc recommendation api calls

### PR DESCRIPTION
**Pull request by CaiWillis**
TIS21-5974: Revalidation: Add Error Codes 100 and 160 to list of known GMC errors

### CODE REVIEW

**Comment by CaiWillis** [comment]
Have updated existing tests for checkrecommendationstatus api, but have not bothered with tests for recommendation submission as they'd currently amount to "when message A is received assert message A is received" as there is no elegant error handling present. Can add these tests if there's a strong desire for it.

File: application/src/main/java/uk/nhs/hee/tis/revalidation/entity/GmcResponseCode.java [hunk 1]
```diff
@@ -36,12 +36,22 @@ public enum GmcResponseCode {
   INVALID_OR_UNEXPECTED_DEFER_DATE("89", "Invalid / Unexpected Defer Date"),
   DOCTOR_NOT_SUBJECT_TO_REVALIDATION("90", "Doctor not subject to revalidation"),
   OPEN_RECOMMENDATION_EXISTS("91", "Open recommendation exists"),
+  DOCTOR_NOT_UNDER_NOTICE("92", "Doctor not under Notice"),
+  DOCTOR_REVALIDATION_ON_HOLD("93", "Doctors revalidation is \"On Hold\""),
+  INVALID_RECOMMENDATION_SUBREASON("94", "Invalid Recommendation Subreason"),
+  RO_NOT_LICENSED("95", "Responsible Officer is not licensed"),
+  MISSING_RO("96", "Missing Responsible Officer"),
+  MISSING_RO_RECOMMENDATION_SUBREASONS("97",
+      "Missing RoRecommendationSubreasons (required when RoRecommendationReason = 1)"),
```
Comment (hihilary) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/entity/GmcResponseCode.java:45]: May I ask if those errors will be exposed to admin user? I don't think they're all user friendly.

Comment (CaiWillis) [reply application/src/main/java/uk/nhs/hee/tis/revalidation/entity/GmcResponseCode.java:45]: Quick look at admins UI seems like they aren't, they'll just be logged in our system (& the method wont crash)

**Review by hihilary — APPROVED** [review]

File: application/src/main/java/uk/nhs/hee/tis/revalidation/entity/GmcResponseCode.java (hunk 1 above)
Comment (CaiWillis) [reply application/src/main/java/uk/nhs/hee/tis/revalidation/entity/GmcResponseCode.java:45]: Ah actually I was wrong, the message is shown to the user, shall I change this one to "Missing RO Recommendation Subreasons"

**Review by hihilary — APPROVED** [review]

**Pull request merged**